### PR TITLE
[memoize-one] add typings for v4.1

### PR DESCRIPTION
**Pull request by vincentbel**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/alexreardon/memoize-one/releases/tag/v4.1.0>>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/memoize-one/memoize-one-tests.ts [hunk 1]
```diff
@@ -3,24 +3,26 @@ import memoizeOne, { EqualityFn } from 'memoize-one';
...
  * Accepts a second argument.
  */
 memoizeOne(add); // $ExpectType (a: number, b: number) => number
 memoizeOne(add, lousyEqualityFn); // $ExpectType (a: number, b: number) => number
 memoizeOne(add, strictEqualityFn); // $ExpectType (a: number, b: number) => number
+memoizeOne(add, equalityFnWithIndex); // $ExpectType (a: number, b: number) => number
 
 /**
- * The second argument can be, but doesn't have to be stritly typed.
+ * The second argument can be, but doesn't have to be strictly typed.
```
Comment (karol-majewski) [inline types/memoize-one/memoize-one-tests.ts]: Nice! 👀

**Review by karol-majewski — COMMENTED** [review]
Nice! Everything looks great.

I would like to ask about one thing though: I know the change introduced in 4.1 is backwards-compatible, but since there is only one version of the typings here, the users that haven't upgraded to v.4 will be left without proper typings.

Perhaps it would be better to leave the old typings be, and [introduce a major version](https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-update-a-package-to-a-new-major-version) so that everyone is happy?

**Comment by VincentBel** [comment]
@karol-majewski hi, I've added the typings for `v3` version back 🤗.

**Comment by karol-majewski** [comment]
@VincentBel Thank you, sir! 🙇

**Pull request merged**